src/components/Personalization/groupDecisions.js

```diff
@@ -86,14 +88,21 @@ const isViewScope = scopeDetails =>
   scopeDetails.characteristics.scopeType &&
   scopeDetails.characteristics.scopeType === VIEW_SCOPE_TYPE;
 
-const extractDecisionsByScope = decisions => {
+const isSurfaceScope = scope => scope.indexOf(WEB_SURFACE) !== -1;
+
+// only exactly matching page surfaces supported in initial AJO Beta release
+const matchesPage = (win, surface) => surface === WEB_SURFACE + getPageLocation(win);
+
+const isMatchingSurfaceScope = (win, scope) => isSurfaceScope(scope) && matchesPage(win, scope);
```

Comment:
The hash part of the page url could have changed in the time between request and processing. Would that be okay?

Reply:
> The hash part of the page url could have changed in the time between request and processing. Would that be okay?

@jonsnyder As the initial AJO Beta release will only support MPA surfaces, for SPA use cases content will be properly rendered only for the first SPA page matching the surface URI. There won't be any re-rendering on subsequent client-side SPA view changes.